packages/actions/test/unit/setup.test.ts

```diff
@@ -0,0 +1,142 @@
+                    fakeCircuitsData.fakeCircuitSmallNoContributors as any
+                ])
+            )
+        })
+        it("should fail when called without being authenticated", async () => {
+            // sign out
+            await signOut(userAuth)
+            assert.isRejected(
+                setupCeremony(userFunctions, fakeCeremoniesData.fakeCeremonyNotCreated, ceremonyPostfix, [
+                    fakeCircuitsData.fakeCircuitSmallNoContributors as any
+                ])
+            )
+        })
+    })
+    describe("extractCircuitMetadata", () => {
+        it("should correctlty extract the circuit metadata", () => {
+            const { curve, wires, constraints, privateInputs, publicInputs, labels, outputs, pot } =
+                extractCircuitMetadata(filePath)
+            expect(curve.trimEnd()).to.be.eq("bn-128")
+            expect(wires).to.be.eq(6)
+            expect(constraints).to.be.eq(1)
+            expect(privateInputs).to.be.eq(3)
+            expect(publicInputs).to.be.eq(1)
+            expect(labels).to.be.eq(8)
+            expect(outputs).to.be.eq(1)
+            expect(pot).to.be.eq(computeSmallestPowersOfTauForCircuit(constraints, outputs))
+        })
+    })
+    describe("getCircuitMetadataFromR1csFile", () => {
+        it("should correctly parse the metadata from the r1cs file", () => {
+            // Extract info from file.
+            expect(extractR1CSInfoValueForGivenKey(filePath, /Curve: .+\n/s).trimEnd()).to.be.eq("bn-128")
+            expect(Number(extractR1CSInfoValueForGivenKey(filePath, /# of Wires: .+\n/s))).to.be.eq(6)
+            expect(Number(extractR1CSInfoValueForGivenKey(filePath, /# of Constraints: .+\n/s))).to.be.eq(1)
+            expect(Number(extractR1CSInfoValueForGivenKey(filePath, /# of Private Inputs: .+\n/s))).to.be.eq(3)
+            expect(Number(extractR1CSInfoValueForGivenKey(filePath, /# of Public Inputs: .+\n/s))).to.be.eq(1)
+            expect(Number(extractR1CSInfoValueForGivenKey(filePath, /# of Labels: .+\n/s))).to.be.eq(8)
+            expect(Number(extractR1CSInfoValueForGivenKey(filePath, /# of Outputs: .+\n/s))).to.be.eq(1)
+        })
+        it("should throw when looking for non-existent metadata", () => {
+            expect(() => extractR1CSInfoValueForGivenKey(filePath, /# of W1res: .+\n/)).to.throw(
+                "Unable to retrieve circuit metadata. Possible causes may involve an error while using the logger. Please, check whether the corresponding `.log` file is present in your local `output/setup/metadata` folder. In any case, we kindly ask you to terminate the current session and repeat the process."
+            )
+        })
+    })
+    describe("estimatePoT", () => {
+        it("should correctly estimate PoT given the number of constraints", () => {
+            expect(computeSmallestPowersOfTauForCircuit(10e6, 2)).to.be.eq(24)
+        })
+    })
+
+    afterAll(async () => {
+        // Clean ceremony and user from DB.
```

Comment:
Did you forget to clean the setupCeremony docs there?

Reply:
Yes, will implement a way to get the ID and delete.